agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -19,75 +19,172 @@ import (
+type TaskProtectionClientFactory struct {
+	Region             string
+	Endpoint           string
+	AcceptInsecureCert bool
 }
 
-// PutTaskProtectionHandler returns an HTTP request handler function for
-// PutTaskProtection API
-func PutTaskProtectionHandler(state dockerstate.TaskEngineState,
-	cluster string) func(http.ResponseWriter, *http.Request) {
+// UpdateTaskProtectionHandler returns an HTTP request handler function for
+// UpdateTaskProtection API
+func UpdateTaskProtectionHandler(state dockerstate.TaskEngineState,
+	credentialsManager credentials.Manager, factory TaskProtectionClientFactoryInterface, cluster string) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		putTaskProtectionRequestType := "api/v1/PutTaskProtection"
+		updateTaskProtectionRequestType := "api/UpdateTaskProtection/v1"
 
 		var request TaskProtectionRequest
 		jsonDecoder := json.NewDecoder(r.Body)
 		jsonDecoder.DisallowUnknownFields()
 		if err := jsonDecoder.Decode(&request); err != nil {
-			logger.Error("PutTaskProtection: failed to decode request", logger.Fields{
+			logger.Error("UpdateTaskProtection: failed to decode request", logger.Fields{
 				loggerfield.Error: err,
 			})
 			writeJSONResponse(w, http.StatusBadRequest,
-				"Failed to decode request", putTaskProtectionRequestType)
+				"Failed to decode request", updateTaskProtectionRequestType)
 			return
 		}
 
 		if request.ProtectionEnabled == nil {
 			writeJSONResponse(w, http.StatusBadRequest,
 				"Invalid request: does not contain 'ProtectionEnabled' field",
-				putTaskProtectionRequestType)
+				updateTaskProtectionRequestType)
 			return
 		}
 
 		taskProtection, err := types.NewTaskProtection(*request.ProtectionEnabled, request.ExpiresInMinutes)
 		if err != nil {
 			writeJSONResponse(w, http.StatusBadRequest,
 				fmt.Sprintf("Invalid request: %v", err),
-				putTaskProtectionRequestType)
+				updateTaskProtectionRequestType)
 			return
 		}
 
 		task, responseCode, err := getTaskFromRequest(state, r)
 		if err != nil {
-			writeJSONResponse(w, responseCode, err.Error(), putTaskProtectionRequestType)
+			writeJSONResponse(w, responseCode, err.Error(), updateTaskProtectionRequestType)
 			return
 		}
 
-		// TODO: Call ECS
-		logger.Info("PutTaskProtection endpoint was called", logger.Fields{
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), updateTaskProtectionRequestType)
+			return
+		}
+
+		logger.Info("UpdateTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster:        cluster,
 			loggerfield.TaskARN:        task.Arn,
 			loggerfield.TaskProtection: taskProtection,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", putTaskProtectionRequestType)
+
+		response, err := ecsClient.UpdateTaskProtection(&ecs.UpdateTaskProtectionInput{
```

Comment:
nit: we might want to call this clusterName to differentiate it from any other identifiers -- and to link it to what we have in our existing cluster API https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Cluster.html.

We have `clusterName` and `clusterArn` but no "cluster".

Reply:
Cool. Will do.

Reply:
Actually, ECS TaskProtection APIs specifies the parameter `cluster` and both `clusterName` and `clusterArn` works for this case. I don't think it is necessary to change it..